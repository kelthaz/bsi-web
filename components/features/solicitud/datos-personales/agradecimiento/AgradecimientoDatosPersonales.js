import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { resetDatosPersonales } from '../../../../../redux/actions/solicitud';
import Agradecimiento from '../../shared/agradecimiento/Agradecimiento';

const AgradecimientoDatosPersonales = () => {
  const { datosPersonales } = useSelector((state) => state.solicitud);
  const dispatch = useDispatch();

  useEffect(
    () => () => {
      dispatch(resetDatosPersonales());
    },
    []
  );

  return (
    <Agradecimiento
      title={
        datosPersonales.enListaNegra
          ? `¡Gracias por tu tiempo ${datosPersonales.primerNombre}!`
          : `¡Gracias, ${datosPersonales.primerNombre}!`
      }
    >
      <p className="body2 text-xs-center text-md-left color-dark-gray sub">
        {datosPersonales.enListaNegra
          ? 'Luego de revisar muy detalladamente los datos que subiste a nuestra plataforma, hemos llegado a la conclusión de que en este momento, no podemos otorgarte el crédito que solicitaste.'
          : 'Por favor revisa tu correo electrónico, te hemos enviado un enlace para verificar y activar tu cuenta.'}
      </p>
    </Agradecimiento>
  );
};
export default AgradecimientoDatosPersonales;
