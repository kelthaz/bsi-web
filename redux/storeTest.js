import { applyMiddleware, createStore } from 'redux';
import thunkMiddleware from 'redux-thunk';
import reducers from './reducers/rootReducer';

const storeTest = () => {
  const store = createStore(reducers, applyMiddleware(thunkMiddleware));
  return store;
};

export default storeTest;
