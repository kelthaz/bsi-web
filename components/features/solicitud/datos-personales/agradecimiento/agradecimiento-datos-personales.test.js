import '@testing-library/jest-dom';
import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import storeTest from '../../../../../redux/storeTest';
import AgradecimientoDatosPersonales from './AgradecimientoDatosPersonales';

describe('Pruebas en el componente AgradecimientoDatosPersonales', () => {
  test('Debe de mostrarse correctamente el componente', () => {
    const store = storeTest();

    const wrapper = mount(
      <Provider store={store}>
        <AgradecimientoDatosPersonales />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe de mostrarse el mensaje de agradecimiento con el nombre', () => {
    // arrange
    const store = storeTest();
    const nombreEsperado = 'SERGIO';
    const {
      solicitud: { datosPersonales },
    } = store.getState();
    datosPersonales.primerNombre = nombreEsperado;

    const wrapper = mount(
      <Provider store={store}>
        <AgradecimientoDatosPersonales />
      </Provider>
    );
    // act
    const h2 = wrapper.find('h2').text();
    // assert
    expect(h2.includes(nombreEsperado)).toBe(true);
  });
});
