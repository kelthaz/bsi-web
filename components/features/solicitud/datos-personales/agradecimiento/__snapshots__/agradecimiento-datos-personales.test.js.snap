// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente AgradecimientoDatosPersonales Debe de mostrarse correctamente el componente 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <AgradecimientoDatosPersonales>
    <div
      className="contedor-fixed"
    >
      <div
        className="contedor-solicitud "
      >
        <div
          className="container"
        >
          <div
            className="row"
          >
            <div
              className="col-md-10"
            >
              <div
                className="d-block d-md-none col-md-2 col-xs-12 text-xs-center mt-4"
              >
                <SvgOk>
                  <svg
                    fill="none"
                    height={137}
                    viewBox="0 0 137 137"
                    width={137}
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <ellipse
                      cx={59.553}
                      cy={59.371}
                      fill="#F4F6F9"
                      rx={59.553}
                      ry={59.371}
                    />
                    <circle
                      cx={79.742}
                      cy={79.865}
                      fill="url(#prefix__paint0_linear)"
                      opacity={0.5}
                      r={56.313}
                    />
                    <path
                      d="M92.326 89.564c1.025-3.573 1.2-7.326.668-10.993l-5.333-36.57c-.683-4.682-5.298-6.36-8.994-4.177M86.53 116.131l.064-14.205 3.981-8.166a26.232 26.232 0 001.63-4.238"
                      stroke="#225AA7"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeMiterlimit={10}
                      strokeWidth={4}
                    />
                    <path
                      d="M83.389 58.319l-7.452-27.38c-2.131-6.415-8.312-8.528-12.786-4.299"
                      stroke="#81C1EA"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeMiterlimit={10}
                      strokeWidth={4}
                    />
                    <path
                      d="M73.623 59.054L62.06 22.983c-1.376-4.298-6.764-5.65-10.04-2.518a6.185 6.185 0 00-1.664 6.203l7.868 26.796"
                      stroke="#81C1EA"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeMiterlimit={10}
                      strokeWidth={4}
                    />
                    <path
                      d="M41.23 90.41a56.56 56.56 0 007.636 8.666l3.32 3.066-.074 13.915 34.438.022"
                      stroke="#225AA7"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeMiterlimit={10}
                      strokeWidth={4}
                    />
                    <path
                      d="M65.68 56.2l-9.114-3.598a12.064 12.064 0 00-11.102 1.186l-9.765 6.483c-3.183 2.113-2.598 7.04 1.028 8.175 2.21.69 4.996.694 8.322-.957 1.004-.498 2.05-.898 3.145-1.135 2.887-.627 7.86-.954 10.41 3.26 1.738 2.869 1.038 6.6-1.51 8.803-1.431 1.236-3.339 2.2-5.665 1.825-1.733-.28-3.249-1.309-4.285-2.729-1.341-1.843-4.103-5.15-7.49-6.5-3.84-1.531-7.538 2.626-5.71 6.35l2.554 5.202a56.461 56.461 0 004.788 7.995"
                      stroke="#81C1EA"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeMiterlimit={10}
                      strokeWidth={4}
                    />
                    <path
                      d="M11.438 45.47l.04 15.097M6.586 53.037l9.748-.04M100.895 9.984l-.016 23.596M93.266 21.79l15.235-.012"
                      stroke="#FFBE12"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeMiterlimit={10}
                      strokeWidth={4}
                    />
                    <defs>
                      <linearGradient
                        gradientUnits="userSpaceOnUse"
                        id="prefix__paint0_linear"
                        x1={133.68}
                        x2={96.55}
                        y1={-17.225}
                        y2={140.362}
                      >
                        <stop
                          stopColor="#fff"
                        />
                        <stop
                          offset={1}
                          stopColor="#B5C8E9"
                        />
                      </linearGradient>
                    </defs>
                  </svg>
                </SvgOk>
              </div>
              <h2
                className="text-xs-center text-md-left color-blue-storm mt-5"
              >
                ¡Gracias, 
                JESUS
                !
              </h2>
              <p
                className="body2 text-xs-center text-md-left color-dark-gray sub"
              >
                Por favor revisa tu correo electrónico, te hemos enviado un enlace para verificar y activar tu cuenta.
              </p>
            </div>
            <div
              className="d-none d-md-block  col-md-2 col-xs-12 text-xs-center mt-4"
            >
              <SvgOk>
                <svg
                  fill="none"
                  height={137}
                  viewBox="0 0 137 137"
                  width={137}
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <ellipse
                    cx={59.553}
                    cy={59.371}
                    fill="#F4F6F9"
                    rx={59.553}
                    ry={59.371}
                  />
                  <circle
                    cx={79.742}
                    cy={79.865}
                    fill="url(#prefix__paint0_linear)"
                    opacity={0.5}
                    r={56.313}
                  />
                  <path
                    d="M92.326 89.564c1.025-3.573 1.2-7.326.668-10.993l-5.333-36.57c-.683-4.682-5.298-6.36-8.994-4.177M86.53 116.131l.064-14.205 3.981-8.166a26.232 26.232 0 001.63-4.238"
                    stroke="#225AA7"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeMiterlimit={10}
                    strokeWidth={4}
                  />
                  <path
                    d="M83.389 58.319l-7.452-27.38c-2.131-6.415-8.312-8.528-12.786-4.299"
                    stroke="#81C1EA"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeMiterlimit={10}
                    strokeWidth={4}
                  />
                  <path
                    d="M73.623 59.054L62.06 22.983c-1.376-4.298-6.764-5.65-10.04-2.518a6.185 6.185 0 00-1.664 6.203l7.868 26.796"
                    stroke="#81C1EA"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeMiterlimit={10}
                    strokeWidth={4}
                  />
                  <path
                    d="M41.23 90.41a56.56 56.56 0 007.636 8.666l3.32 3.066-.074 13.915 34.438.022"
                    stroke="#225AA7"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeMiterlimit={10}
                    strokeWidth={4}
                  />
                  <path
                    d="M65.68 56.2l-9.114-3.598a12.064 12.064 0 00-11.102 1.186l-9.765 6.483c-3.183 2.113-2.598 7.04 1.028 8.175 2.21.69 4.996.694 8.322-.957 1.004-.498 2.05-.898 3.145-1.135 2.887-.627 7.86-.954 10.41 3.26 1.738 2.869 1.038 6.6-1.51 8.803-1.431 1.236-3.339 2.2-5.665 1.825-1.733-.28-3.249-1.309-4.285-2.729-1.341-1.843-4.103-5.15-7.49-6.5-3.84-1.531-7.538 2.626-5.71 6.35l2.554 5.202a56.461 56.461 0 004.788 7.995"
                    stroke="#81C1EA"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeMiterlimit={10}
                    strokeWidth={4}
                  />
                  <path
                    d="M11.438 45.47l.04 15.097M6.586 53.037l9.748-.04M100.895 9.984l-.016 23.596M93.266 21.79l15.235-.012"
                    stroke="#FFBE12"
                    strokeLinecap="round"
                    strokeLinejoin="round"
                    strokeMiterlimit={10}
                    strokeWidth={4}
                  />
                  <defs>
                    <linearGradient
                      gradientUnits="userSpaceOnUse"
                      id="prefix__paint0_linear"
                      x1={133.68}
                      x2={96.55}
                      y1={-17.225}
                      y2={140.362}
                    >
                      <stop
                        stopColor="#fff"
                      />
                      <stop
                        offset={1}
                        stopColor="#B5C8E9"
                      />
                    </linearGradient>
                  </defs>
                </svg>
              </SvgOk>
            </div>
          </div>
        </div>
      </div>
    </div>
  </AgradecimientoDatosPersonales>
</Provider>
`;
